demos/python_demos/human_pose_estimation_3d_demo/pose_extractor/wrapper.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include "numpy/arrayobject.h"
+
+#include <opencv2/core/core.hpp>
+
+#include "extract_poses.hpp"
+
+static std::vector<cv::Mat> wrap_feature_maps(PyArrayObject* py_feature_maps) {
+    int num_channels = static_cast<int>(PyArray_SHAPE(py_feature_maps)[0]);
+    int h = static_cast<int>(PyArray_SHAPE(py_feature_maps)[1]);
+    int w = static_cast<int>(PyArray_SHAPE(py_feature_maps)[2]);
+    float* data = static_cast<float*>(PyArray_DATA(py_feature_maps));
+    std::vector<cv::Mat> feature_maps(num_channels);
+    for (long c_id = 0; c_id < num_channels; c_id++)
+    {
+        feature_maps[c_id] = cv::Mat(h, w, CV_32FC1,
+                                     data + c_id * PyArray_STRIDE(py_feature_maps, 0) / sizeof(float),
+                                     PyArray_STRIDE(py_feature_maps, 1));
+    }
+    return feature_maps;
+}
+
+static PyObject* extract_poses(PyObject* self, PyObject* args) {
+    PyArrayObject* py_heatmaps;
+    PyArrayObject* py_pafs;
+    int ratio;
+    if (!PyArg_ParseTuple(args, "OOi", &py_heatmaps, &py_pafs, &ratio))
+    {
+        throw std::runtime_error("passed non-numpy array as argument");
```

Comment:
You can't throw a C++ exception from a Python function. You need to raise a Python exception.

Reply:
Correction: `PyArg_ParseTuple` will raise an exception by itself if it fails. So all you need to do is return `nullptr`.